wp/plugin/tests/test_libs/test_wp.ml

```diff
@@ -0,0 +1,577 @@
+      ~reg_list:(lift_out_regs models)
+      ~checker:(check_list models |> Some);
+  );
+
+  "Loop incomplete unroll"         >:: test_skip fail_msg
+    (
+      let models = [ [("RDI", "0x0000000000000005")]; ] in
+      test_plugin "loop/loop_depth_one" sat ~script:"run_wp_less_loop.sh"
+        ~reg_list:(lift_out_regs models)
+        ~checker:(check_list models |> Some)
+    );
+
+  "Loop incomplete unroll"         >:: test_skip fail_msg
+    (test_plugin "loop/loop_depth_one" sat);
+
+  "Nested function calls"               >: test_plugin "nested_function_calls" unsat;
+
+  "Nested function calls: inline regex" >: (
+    let models = [ [("RDI", "0x0000000000000004")]; ] in
+    test_plugin "nested_function_calls" sat
+      ~script:"run_wp_inline_regex.sh" ~reg_list:(lift_out_regs models)
+      ~checker:(check_list models |> Some)
+  );
+
+  "Nested function calls: inline all"   >: (
+    let models = [ [("RDI", "0x0000000000000004")]; ] in
+    test_plugin "nested_function_calls" sat
+      ~script:"run_wp_inline_all.sh" ~reg_list:(lift_out_regs models)
+      ~checker:(check_list models |> Some)
+  );
+
+
+  "Nested ifs"                     >: test_plugin "nested_ifs" unsat;
+  "Nested ifs: goto"               >: test_plugin "nested_ifs" unsat
+    ~script:"run_wp_goto.sh";
+  "Nested ifs: inline"             >: test_plugin "nested_ifs" unsat
+    ~script:"run_wp_inline.sh";
+
+  "Null dereference: no check"     >: test_plugin "null_deref" unsat;
+  "Null dereference: with check"   >: test_plugin "null_deref" sat
+    ~script:"run_wp_null_deref.sh";
+
+  "User defined postcondition"     >: test_plugin "return_argc" sat;
+
+  "Simple WP"                      >: (
+    let models = [[("RDI", "0x0000000000000003")]] in
+    test_plugin "simple_wp" sat ~reg_list:(lift_out_regs models)
+      ~checker:(check_list models |> Some)
+  );
+
+  "Simple WP: precondition"        >: test_plugin "simple_wp" unsat
+    ~script:"run_wp_pre.sh";
+
+  "Single test without pointer flag" >: test_plugin "pointer_flag_single"
+    sat ~script:"run_wp_sat.sh";
+  "Single test with pointer flag"    >: test_plugin "pointer_flag_single"
+    unsat ~script:"run_wp_unsat.sh";
+
+  "Verifier assume SAT"            >: test_plugin "verifier_calls" sat
+    ~script:"run_wp_assume_sat.sh";
+  "Verifier assume UNSAT"          >: test_plugin "verifier_calls" unsat
+    ~script:"run_wp_assume_unsat.sh";
+  "Verifier nondent"               >: test_plugin "verifier_calls" sat
+    ~script:"run_wp_nondet.sh";
+]
+
+let nqueens_tests = List.range 4 20 |> List.map ~f:(fun i ->
+    let description = Printf.sprintf "NQueens solver %dx%d" i i in
+    let script = Printf.sprintf "run_wp.sh" in
+    (description >: test_plugin "nqueens" sat
+       ~reg_list:(get_register_args 6 `x86_64 |> StringSet.of_list)
+       ~checker:(check_n_queens i `x86_64 |> Some)
+       ~script:script ~args:[string_of_int i]))
+
+let integration_tests = List.append [
```

Comment:
This file is getting rather large. What are your thoughts on separating the unit tests and integration tests into two different files?